masonry/src/passes/update.rs

```diff
@@ -416,17 +416,14 @@ fn update_new_widgets(
     }
     state.item.children_changed = false;
 
-    // This will recursively call WidgetPod::lifecycle for all children of this widget,
-    // which will add the new widgets to the arena.
     {
-        let mut ctx = LifeCycleCtx {
-            global_state,
-            widget_state: state.item,
+        let mut ctx = RegisterCtx {
             widget_state_children: state.children.reborrow_mut(),
             widget_children: widget.children.reborrow_mut(),
         };
-        let event = LifeCycle::Internal(crate::InternalLifeCycle::RouteWidgetAdded);
-        widget.item.lifecycle(&mut ctx, &event);
+        // The widget will call `RegisterCtx::register_child` on all its children,
+        // which will add the new widgets to the arena.
+        widget.item.register_children(&mut ctx);
     }
```

Comment:
We should mention this method at:
https://github.com/linebender/xilem/blob/12ddfa35a556fa426f0bbe9c58fe9a7792bbae9d/masonry/src/passes/mod.rs#L31

Reply:
I'm leaning more towards adding a safety check in the update_new_widgets pass. If the error you mention still happens after that safety check, then the problem likely comes from the widget's children_ids implementation, not register_children.